### ISSUE TITLE
How will ML be embedded in Photoshop (Mac & Win)?

### ISSUE THREAD

**Issue by timriot** [kilianc/adobe-generator#60]
I will speak to Tom R, John P, & Jeffrey C to see what our options are.

**Comment by kilianc** [comment]
Also, investigate how node-canvas compiles on windows.

## 

Kilian C.

**Comment by timriot** [comment]
Looks like it works!

https://github.com/LearnBoost/node-canvas/wiki/Installation---Windows

**Comment by timriot** [comment]
I'm going to rename this issue to check out node-canvas and assign this to
you. Is that cool?

**Comment by kilianc** [comment]
Yeah, it has to, otherwise start searching the highest bridge around there and make the jump.

It's cool to rename the issue, but we need to talk with some build guy,
I don't know how this will be integrated/compiled and what are requirements on windows
to support this library (dynamic libs for examples)

## 

Kilian C.

**Comment by kilianc** [comment]
How was [cairo](http://www.cairographics.org/) license?

## 

Kilian C.

**Comment by timriot** [comment]
I didn't hear anything bad about the GPL2.1 license associated with cairo,
so I am assuming it's ok. I will follow up with the legal contact I have.

-Tim

**Comment by timriot** [comment]
Sounds like it makes more sense for me to keep hold of this issue. I will
talk to a build dude about compiling/embedding all this crap.

**Comment by kilianc** [comment]
any updates?

**Comment by timriot** [comment]
No updates, yet. Issues getting Cairo on dev machines, has kept this from
progressing. I will continue to work with John, Tom to get to a good place
this week.

**Comment by kilianc** [comment]
uhm. compiling cairo is an issue? we just need the dlyb.
mac or pc?

## 

Kilian C.

**Comment by timriot** [comment]
Compiling cairo is part of the issue. Even with the dylib / instructions, I
couldn't get canvas to install without Cairo being compiled first. =\

-Tim

**Comment by kilianc** [comment]
Of course. read here: https://github.com/kilianc/motherlover/blob/develop/Motherlover/Resources/server/README.md

## 

Kilian C.

**Comment by timriot** [comment]
Yeah, I followed these instructions and couldn't get it to work properly.
I'll spend some more time with in later today. Meanwhile, we will need to
embed node, as well.

-Tim

**Comment by timriot** [comment]
Tried again.

I got the script working after I chased down all of the following dylibs
and put 'em in my ./dylib directory:

libX11.6.dylib
libXau.6.dylib
libXdmcp.6.dylib
libXrender.1.dylib
libcairo.2.dylib
libfontconfig.1.dylib
libfreetype.6.dylib
libjpeg.8.dylib
libpixman-1.0.dylib
libpng15.15.dylib
libxcb-render.0.dylib
libxcb-shm.0.dylib
libxcb.1.dylib

Stupid cairo and all it's dependencies. Glad we're getting rid of the
sucker.

-Tim
